### ISSUE TITLE
Segfault in DCDM calculations

### ISSUE THREAD

**Issue by fmccarthy** [lesgourg/class_public#515]
I find a segfault in classy v3.2.0 for certain configurations of input parameters when asking for DCDM  output. For example the following set of parameters:

A_s = 2.1080159136965503e-09
n_s = 0.9634998212853542
omega_b = 0.022299630125110836
omega_cdm =  0.11732130667401973
m_ncdm = 0.06
tau_reio = 0.06087141721157738
100*theta_s =  1.0419379601460101
nonlinear_min_k_max = 20
N_ncdm =  1
N_ur = 2.0328
output =  tCl pCl tCl, lCl lCl, mPk pCl, dCl
modes =  s
non_linear =  hmcode
l_max_scalars =  2500
lensing = yes
P_k_max_1/Mpc =  1

Gamma_dcdm = 0.8914400255168418
Omega_ini_dcdm = 0.006748518551475872

Produces a segfault (I think this occurs somewhere in the "shooting" algorithm)

but configurations with parameters slightly perturbed from this seem to work fine. Not sure what's going on. Thanks :)

**Comment by ShazAlvi** [comment]
Did you try to remove the nonlinear computation and instead use linear computation?

**Comment by fmccarthy** [comment]
Yes, that doesn't help; I don't think the shooting algorithm should know about the nonlinearities in any case

**Comment by fmccarthy** [comment]
As an update: I'm avoiding this for now by reverting to class v2.10

**Comment by kcroker** [comment]
Could you please run this with gdb and get the actual location of the segfault?
